src/ResourceManager/AnalysisServices/Commands.AnalysisServices/Models/AzureAnalysisServicesFirewallConfig.cs

```diff
@@ -0,0 +1,34 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using Microsoft.Azure.Management.Analysis.Models;
+using System.Collections.Generic;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.Commands.AnalysisServices.Models
+{
+    public class AzureAnalysisServicesFirewallConfig
+    {
+        public bool EnablePowerBIService { get; set; }
+
+        public List<AzureAnalysisServicesFirewallRule> FirewallRules { get; set; }
+
+        public AzureAnalysisServicesFirewallConfig()
+        {
+            EnablePowerBIService = false;
```

Comment:
This will automatically set the values rather than using the parameter values - you need another constructor to take in the parameter values.